### PR TITLE
Add integration with circleci

### PR DESCRIPTION
**Pull request by na-ga**
## About

Added [circleci](https://circleci.com/) to CI/CD integration.

## Ref

CircleCI webhook spec: https://circleci.com/docs/2.0/webhooks/

## Example

Table: events_raw

```
SELECT * FROM `xxx.four_keys.events_raw` WHERE source = "circleci" ORDER BY time_created DESC LIMIT 1

[
  {
    "event_type": "workflow-completed",
    "id": "091b1c93-a0f5-3135-8e39-1e36f5e565aa",
    "metadata": "{\"type\": \"workflow-completed\", \"id\": \"091b1c93-a0f5-3135-8e39-1e36f5e565aa\", \"happened_at\": \"2021-09-10T06:01:57.779Z\", \"webhook\": {\"id\": \"b5091b03-71db-45b4-b283-51bad7440725\", \"name\": \"FourKeys\"}, \"workflow\": {\"id\": \"c5246284-4d2b-439e-b61f-8bfc6dccae9d\", \"name\": \"build_and_deploy_on_fourkeys\", \"created_at\": \"2021-09-10T06:00:08.864Z\", \"stopped_at\": \"2021-09-10T06:01:57.650Z\", \"url\": \"https://app.circleci.com/pipelines/github/org/repo/106/workflows/c5246284-4d2b-439e-b61f-8bfc6dccae9d\", \"status\": \"success\"}, \"pipeline\": {\"id\": \"a34aeeeb-d9ef-47a9-8f76-ae8ecb5f89d8\", \"number\": 106, \"created_at\": \"2021-09-10T06:00:08.155Z\", \"trigger\": {\"type\": \"webhook\"}, \"vcs\": {\"provider_name\": \"github\", \"origin_repository_url\": \"https://github.com/org/repo\", \"target_repository_url\": \"https://github.com/org/repo\", \"revision\": \"1967a468e5ca0a610ea5a64bd5014dfb5859220f\", \"commit\": {\"subject\": \"Fix\", \"body\": \"\", \"author\": {\"name\": \"na-ga\", \"email\": \"xxx@gmail.com\"}, \"authored_at\": \"2021-09-10T06:00:03Z\", \"committer\": {\"name\": \"na-ga\", \"email\": \"xxx@gmail.com\"}, \"committed_at\": \"2021-09-10T06:00:03Z\"}, \"branch\": \"fourkeys\"}}, \"project\": {\"id\": \"3a06f674-3e67-4f37-b5be-e9375864d508\", \"name\": \"repo\", \"slug\": \"github/org/repo\"}, \"organization\": {\"id\": \"45f7dfd7-5dfa-426c-8b7c-fb83570a66d5\", \"name\": \"org\"}}",
    "time_created": "2021-09-10 06:01:57.779 UTC",
    "signature": "v1=cfc29c70bd861c7907a7850c4af2b9a34ac2cd49f342e725908953dbd5a3337c",
    "msg_id": "3016210390420658",
    "source": "circleci"
  },
]
```

Table: deployments

```
SELECT * FROM `xxx.four_keys.deployments` WHERE source = "circleci" AND deploy_id = "091b1c93-a0f5-3135-8e39-1e36f5e565aa" LIMIT 1

[
  {
    "source": "circleci",
    "deploy_id": "091b1c93-a0f5-3135-8e39-1e36f5e565aa",
    "time_created": "2021-09-10 06:01:57.779 UTC",
    "main_commit": "1967a468e5ca0a610ea5a64bd5014dfb5859220f",
    "changes": [
      "1967a468e5ca0a610ea5a64bd5014dfb5859220f"
    ]
  }
]
```

## Outstanding questions

CircleCI workflows may only perform testing and not deployment. Therefore, the query that generates the deployments table (view) targets the success event of the workflow that contains the name `deploy`. Is there any better way to do this?

https://github.com/GoogleCloudPlatform/fourkeys/blob/84b88f4c0a4186978ad0a73ea5939505803de6cd/queries/deployments.sql#L38-L47

### CODE REVIEW

**Comment by davidstanke** [comment]
Hi @na-ga ! Thanks for the PR! At first glance, this looks great. :) I'd like to get Dina's eyes on it as well, so expect a proper review shortly.

**Comment by dinagraves** [comment]
This is so excellent!  I'm going to test it out with our installer and give more feedback.

**Review by dinagraves — CHANGES_REQUESTED** [review]
> CircleCI workflows may only perform testing and not deployment. Therefore, the query that generates the deployments table (view) targets the success event of the workflow that contains the name `deploy`. Is there any better way to do this?

I think this is a fine starting point.  We have to acknowledge that everyone is going to do it a little different and they'll have to find the right signals to filter out the successful deployments.  

I've tested this out manually and set it up with a CircleCI deployment sending to the Webhook.  I can confirm that it works end-to-end.

Thank you so much for this excellent work!  My only change request is to revert the files back in the other builders.  We don't need the `sys.stdout.flush()` because of how we set up the Dockerfile.

File: bq-workers/github-parser/main.py [hunk 1]
```diff
@@ -15,7 +15,6 @@
 import base64
 import os
 import json
-import sys
```
Comment (dinagraves) [inline bq-workers/github-parser/main.py:18]: Thank you for being so thorough! I really appreciate it!

**Review by dinagraves — APPROVED** [review]

**Comment by dinagraves** [comment]
Thank you again for this contribution!  If you haven't done so yet, feel free to join us on the #fourkeys channel on the [Google Cloud Platform Slack](https://join.slack.com/t/googlecloud-community/shared_invite/zt-m973j990-IMij2Xh8qKPu7SaHfOcCFg)!

**Comment by na-ga** [comment]
@dinagraves @davidstanke 
Thanks for the review and the merge!

**Comment by na-ga** [comment]
Thanks for the invitation to slack. I've just joined.

**Pull request merged**